dlrover/python/elastic_agent/monitor/resource.py

```diff
@@ -53,6 +54,38 @@ def get_used_memory():
     return int(mem.used / 1024 / 1024)
 
 
+def get_gpu_stats():
+    """ "Get the used gpu info of the container"""
+    device_count = pynvml.nvmlDeviceGetCount()
+    gpu_stats_list = []
+    for i in range(device_count):
+        handle = pynvml.nvmlDeviceGetHandleByIndex(i)
+        memory_info = pynvml.nvmlDeviceGetMemoryInfo(handle)
+        total_memory = memory_info.total / (1024**2)
+        used_memory = memory_info.used / (1024**2)
+
+        # Get GPU utilization
+        utilization = pynvml.nvmlDeviceGetUtilizationRates(handle)
+        gpu_utilization = utilization.gpu
+
+        gpu_stats_list.append(
+            {
```

Comment:
It is better to wrap the dictionary to a dataclasss because the dataclass is more readable and reusable than a dict.

```
from dataclasses import dataclass
 
@dataclass
class GPUMetric:
    index: int
    total_memory_mb: int
    used_mmeory_mb: int
    gpu_utilization: float
```

Reply:
Rewrited using dataclass.